[3006.x] fix inline pillar being saved to pillarcache when it shouldn't be

whytewolf opened the pull request:
### What does this PR do?

stops the pillar_override from being saved when saving the cache file. but other wise is included in pillar rendering.

### What issues does this PR fix or reference?
Fixes #66292 

### Previous Behavior
inline_pillar being saved in cache

### New Behavior
inline_pillar not saved in cachet

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

Commit:
add fix for pillar_override being saved in PillarCache

Commit:
add changelog and tests

jtraub91 commented on salt/pillar/__init__.py:
```diff
@@ -460,20 +460,23 @@ def _minion_cache_path(self, minion_id):
         return os.path.join(self.opts["cachedir"], "pillar_cache", minion_id)
 
-    def fetch_pillar(self):
+    def fetch_pillar(self, save_override=True):
```
Why does `save_override` default to `True`? Why would we ever want to cache the pillar override?

dwoz replied:
I wonder about this too, do we even need the `save_override` option?

whytewolf replied:
fetch_pillar isn't what does the saving,its return is what gets saved. and as such i wanted to make sure if it is being called elsewhere from some other source either in salt or third party module it retained the same functionality it had.

whytewolf replied:
i can remove that. it only gets called twice in salt. will simplify the test a little bit as i can remove the second check.

jtraub91 replied:
I'm mainly just curious why `save_override=False` doesn't need to be applied to line 520 as well
https://github.com/whytewolf/salt/blob/pillar_cache/salt/pillar/__init__.py#L520

whytewolf replied:
that one missing is a mistake. I"ll talk with @dwoz this morning, see if we want to skip the backwards compatibility. if we do I"ll just strip out the save_override. otherwise I'll fix that line.

whytewolf replied:
on second thought, no one should be calling PillarCache directly anyway. I'll just strip out the save_override. save the headache of correcting it later.

Commit:
remove backwords compatibility.

internally salt doesn't need it. and if someone is using PillarCache
incorrectly that is on them.

Commit:
Merge remote-tracking branch 'upstream/3006.x' into pillar_cache

jtraub91 reviewed (approved):
lgtm
